ml-agents/mlagents/trainers/sac/policy.py

```diff
@@ -0,0 +1,318 @@
+        trainer_params: Dict[str, Any],
+        reward_signal_configs: Dict[str, Any],
+        is_training: bool,
+        load: bool,
+        seed: int,
+    ) -> None:
+        with self.graph.as_default():
+            self.model = SACModel(
+                brain,
+                lr=float(trainer_params["learning_rate"]),
+                h_size=int(trainer_params["hidden_units"]),
+                init_entcoef=float(trainer_params["init_entcoef"]),
+                max_step=float(trainer_params["max_steps"]),
+                normalize=trainer_params["normalize"],
+                use_recurrent=trainer_params["use_recurrent"],
+                num_layers=int(trainer_params["num_layers"]),
+                m_size=self.m_size,
+                seed=seed,
+                stream_names=list(reward_signal_configs.keys()),
+                tau=float(trainer_params["tau"]),
+                gammas=list(_val["gamma"] for _val in reward_signal_configs.values()),
+                vis_encode_type=trainer_params["vis_encode_type"],
+            )
+            self.model.create_sac_optimizers()
+
+        self.inference_dict.update(
+            {
+                "action": self.model.output,
+                "log_probs": self.model.all_log_probs,
+                "value_heads": self.model.value_heads,
+                "value": self.model.value,
+                "entropy": self.model.entropy,
+                "learning_rate": self.model.learning_rate,
+            }
+        )
+        if self.use_continuous_act:
+            self.inference_dict["pre_action"] = self.model.output_pre
+        if self.use_recurrent:
+            self.inference_dict["memory_out"] = self.model.memory_out
+        if (
+            is_training
+            and self.use_vec_obs
+            and trainer_params["normalize"]
+            and not load
+        ):
+            self.inference_dict["update_mean"] = self.model.update_normalization
+
+        self.update_dict.update(
+            {
+                "value_loss": self.model.total_value_loss,
+                "policy_loss": self.model.policy_loss,
+                "q1_loss": self.model.q1_loss,
+                "q2_loss": self.model.q2_loss,
+                "entropy_coef": self.model.ent_coef,
+                "entropy": self.model.entropy,
+                "update_batch": self.model.update_batch_policy,
+                "update_value": self.model.update_batch_value,
+                "update_entropy": self.model.update_batch_entropy,
+            }
+        )
+
+    def create_reward_signals(self, reward_signal_configs: Dict[str, Any]) -> None:
+        """
+        Create reward signals
+        :param reward_signal_configs: Reward signal config.
+        """
+        self.reward_signals: Dict[str, RewardSignal] = {}
+        with self.graph.as_default():
+            # Create reward signals
+            for reward_signal, config in reward_signal_configs.items():
+                if type(config) is dict:
```

Comment:
Why would the type of the config not be a dict?

Reply:
This is kind of a hack. It's because in SAC we have an additional `reward_signal` hyperparameters `num_update` that can be set to do multiple reward signals updates, in which case the config would be an `int`. 

Open to cleaner ways of doing this.

Reply:
num_update could just be a member variable on the reward signal, and be explicitly checked when updating.  I'd prefer it work that way if possible -- though it's OK if you want to push that to a follow-on PR / next release.

Reply:
That seems like a minor change - I'll change it when I get a chance

Reply:
I realized the `num_update` would break the `reward_signal` abstraction that we created in the parallel update refactor - need to think about this in more depth.

Reply:
Well, I don't love this solution but it's OK for not.